### DISCUSSION TITLE
What is the difference between 'git pull' and 'git fetch'?

### DISCUSSION THREAD

**Discussion by senior-software-engineers** [community/community#135559] [Pull Requests] [answered]
### Select Topic Area

Question

### Body

What are the differences between [git pull](https://git-scm.com/docs/git-pull) and [git fetch](https://git-scm.com/docs/git-fetch)?

**Comment by everforyou** [comment, marked as answer]
In the simplest terms, [git pull](http://git-scm.com/docs/git-pull) does a [git fetch](http://git-scm.com/docs/git-fetch) followed by a [git merge](http://git-scm.com/docs/git-merge).

git fetch updates your remote-tracking branches under refs/remotes/<remote>/. This operation is safe to run at any time since it never changes any of your local branches under refs/heads.

git pull brings a local branch up-to-date with its remote version, while also updating your other remote-tracking branches.

From the Git documentation for [git pull](http://git-scm.com/docs/git-pull):

**Comment by LavishT04** [comment]
Git fetch and git pull serve different purposes in managing remote repository changes. Git fetch downloads new data from a remote repository, updating your remote-tracking branches without modifying your working directory or local branches. It's a safe way to see changes without immediately integrating them. On the other hand, git pull combines fetch and merge operations. It downloads changes from the remote repository and automatically merges them into your current local branch, updating your working directory. While git pull is more convenient for quick updates, git fetch offers more control, allowing you to review changes before deciding how to integrate them.
